coding-challenges.js

```diff
@@ -2,59 +2,111 @@
 // Expected output example2 (can be a different order): ["blue", "green", "pink"]
 // Expected output example3 (can be a different order): ["pink", "green"]
 
-
-// --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
+var collections = ["purple", "blue", "green", "yellow", "pink"]
+removeAndShuffle(collections)
+console.log(collections)
+removeAndShuffle(collections)
+console.log(collections)
+removeAndShuffle(collections)
+console.log(collections)
 
 
 
+// --------------------2) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
+const cubedSum= (nums) =>{
+    //create a varible to store the result
+    let sum = 0
+    //add each cubed number to the sum
+    nums.map(value => {
+        sum += value**3
+    })
+    return sum
+}
 
 
 var cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 var cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
-
-
+console.log(cubedSum(cubeAndSum1))
+console.log(cubedSum(cubeAndSum2))
 
 // --------------------3) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
-
-
+const minAndMax = (nums) =>{
+    //create two variables to store max and min, starting from the first element 
+    let min = nums[0]
+    let max = nums[0]
+    //loop through the array, if found a number is less than the current min, set min to that number, and if the number is larger than current max, set max to that number 
+    for(let i = 0; i < nums.length; i++){
+        min = (min > nums[i])?nums[i]:min
+        max = (max < nums[i])?nums[i]:max
+    }
+    return [min, max]
+}
 
 
 var nums1 = [3, 56, 90, -8, 0, 23, 6]
 // Expected output: [-8, 90]
 var nums2 = [109, 5, 9, -59, 8, 24]
 // Expected output: [-59, 109]
+console.log(minAndMax(nums1))
+console.log(minAndMax(nums2))
 
 
 // --------------------4) Create a function that takes in a string and returns a string with every other letter capitalized.
-
-
-
+const capitalize = (string) =>{
+    //create a variable to store the result
+    let finalString = ""
+    //loop through the string, set every other letter capitalized
+    for(let i = 0; i < string.length; i++){
+        finalString += (i%2 == 0)?string[i]:string[i].toUpperCase()
+    }
+    return finalString
+}
 
 
 var testString1 = "albatross"
 // Expected output: "aLbAtRoSs"
 var testString2 = "jabberwocky"
 // Expected output: "jAbBeRwOcKy"
-
+console.log(capitalize(testString1))
+console.log(capitalize(testString2))
 
 // --------------------5) Create a function that takes in two arrays as arguments and returns one array with no duplicate values. STRETCH: Use the spread operator.
 
-
-
-
+const noDupe = (arr1, arr2) =>{
+    //combine 2 arrays, use set to remove duplicates
+    let setArr = new Set([...arr1, ...arr2])
+    //transfer set to array
+    return [...setArr]
+    
+}
```

Comment:
good use of Set. certainly a very efficient solution. just be prepared for an interviewer to ask you to write out the long way. i always suggest considering an optimized solution but for the sake of practice, i would put a "naive" solution (meaning, less efficient or possibly your initial approach), and an "optimized" solution in your assessment. this practice will have a real-world application as well as it is _extremely_ helpful/impressive in an interview setting. showing prospective employers how you approach a problem and how you go about refactoring is very valuable.

Reply:
const noDupe = (arr1, arr2) =>{
    //combine two arrays to a new one
    let wholeArr = [...arr1, ...arr2]
    //create a empty array to receive results
    let finalArr = []
    //loop through the whole array, find those index match the indexOf function,then push into result
    for(let i = 0; i < wholeArr.length; i++){
      if (i == wholeArr.indexOf(wholeArr[i])){
        finalArr.push(wholeArr[i])
      }
    }
    return finalArr
}

I tried another way using indexof to dump the duplicates.